<?xml version="1.0"?>
<launch>

   <!-- <node pkg="tf" type="static_transform_publisher" name="base_link" args="0.3 0.4 0.5 0.6 0.7 0.8 0.9 base_link rslidar 20" />
   <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="1 0 0 0 0 0 1 base_link link1 100" /> -->

  <!-- base_link -->
  <node name="base_link" pkg="testcfg" type="tunable_static_tf_broadcaster_node.py" >
    <param name="rate" value="10.0" type="double" /> <!-- tfの更新レート[Hz]-->
    <param name="header_frame" value="world" type="string" /> <!-- tfのヘッダフレームID -->
    <param name="child_frame" value="base_link" type="string" /> <!-- tfの子フレームID-->
    <param name="yaml" value="$(find testcfg)/params/base_link.yaml" type="string" /> <!-- 初期化用yaml -->
  </node>

  <!-- arm -->
  <node name="arm" pkg="testcfg" type="tunable_static_tf_broadcaster_node.py" >
    <param name="rate" value="10.0" type="double" /> <!-- tfの更新レート[Hz]-->
    <param name="header_frame" value="base_link" type="string" /> <!-- tfのヘッダフレームID -->
    <param name="child_frame" value="arm" type="string" /> <!-- tfの子フレームID-->
    <param name="yaml" value="$(find testcfg)/params/arm.yaml" type="string" /> <!-- 初期化用yaml -->
  </node>

  設定用rqt_reconfigure
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />

  <!-- dynamic_reconfigureの調整済みパラメータロード -->
  <node name="load_base_link" pkg="dynamic_reconfigure" type="dynparam" args="load base_link $(find testcfg)/params/base_link.yaml" />
  <node name="load_arm" pkg="dynamic_reconfigure" type="dynparam" args="load arm $(find testcfg)/params/arm.yaml" />

</launch>
